### PR TITLE
Fix ClientWorker thread waiting issue.

### PR DESCRIPTION
**Pull request by djkevincr**
Related public JIRA : https://wso2.org/jira/browse/ESBJAVA-5071.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyServiceMessageReceiver.java [hunk 1]
```diff
@@ -78,7 +78,18 @@ public void receive(org.apache.axis2.context.MessageContext mc) throws AxisFault
...
+                String[] cids = null;
+                try {
+                    cids = mc.getAttachmentMap().getAllContentIDs();
+                } catch (Exception ex){
+                    //partially read stream could lead to corrupted attachment map and hence this exception
+                    //corrupted attachment map leads to inconsistent runtime exceptions and behavior
+                    //discard the attachment map for the fault handler invocation
+                    //ensure the successful completion for fault handler flow
+                    mc.setAttachmentMap(null);
+                    log.debug("Synapse encountered an exception when reading attachments from bytes stream. " +
```
Comment (Asitha) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyServiceMessageReceiver.java]: Shall we add the exception as well to the log?

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java [hunk 2]
```diff
@@ -305,7 +309,18 @@ private void handleMessage(String messageID ,MessageContext response,
...
+                try {
+                    cids = response.getAttachmentMap().getAllContentIDs();
+                } catch (Exception ex){
+                    //partially read stream could lead to corrupted attachment map and hence this exception
+                    //corrupted attachment map leads to inconsistent runtime exceptions and behavior
+                    //discard the attachment map for the fault handler invocation
+                    //ensure the successful completion for fault handler flow
+                    response.setAttachmentMap(null);
+                    log.debug("Synapse encountered an exception when reading attachments from bytes stream. " +
+                            "Hence Attachments map is dropped from the message context.");
```
Comment (Asitha) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java]: Shall we log the exception as well? BTW shouldn't this be an error log?

File: modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyServiceMessageReceiver.java (hunk 1 above)
Comment (djkevincr) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyServiceMessageReceiver.java]: +1

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java (hunk 2 above)
Comment (djkevincr) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java]: +1

**Pull request merged**